fintrack/src/components/TransactionForm.js

```diff
@@ -0,0 +1,208 @@
+import { useNavigate, useLocation } from 'react-router-dom'
+
+import { createTransaction, updateTransaction, deleteTransaction } from '../utils/api-transactions'
+
+import UserContext from '../context/UserContext'
+import { AuthContext } from '../context/AuthContext'
+
+function TransactionForm() {
+    const navigate = useNavigate()
+    const location = useLocation()
+
+    const { authToken } = useContext(AuthContext)
+    // get the user's categories and transaction types from context
+    const { authUser } = useContext(UserContext)
+
+    const [isEditMode, setIsEditMode] = useState(false)
+    const [account, setAccount] = useState(location.state?.selectedAccount)
+    const [transaction, setTransaction] = useState(location.state.transaction)
+    const [formData, setFormData] = useState({
+        account: '',
+        transactionType: '',
+        category: '',
+        amount: '',
+        description: ''
+    })
+    const [feedbackMessage, setFeedbackMessage] = useState('')
+
+    useEffect(() => {
+        console.log('use effect')
+        
+        const extractData = () => {
+            if (location.state?.transaction) {
+                setTransaction(location.state.transaction)
+                setIsEditMode(true)
+                setFormData({
+                    account: transaction.account._id,
+                    transactionType: transaction.transactionType._id,
+                    category: transaction.category._id,
+                    amount: transaction.amount,
+                    description: transaction.description
+                })
+            } else if (location.state?.selectedAccount) {
+                setAccount(location.state.selectedAccount)
+                setIsEditMode(false)
+                // account change functionality is not yet implemented
+                setFormData({
+                    account: account._id,
+                    transactionType: '',
+                    category: '',
+                    amount: '',
+                    description: ''
+                })
+            }
+        }
+        extractData()
+
+    }, [location, account, transaction])
+
+    
+    const categories = authUser.categories || []
+    const transactionTypes = authUser.transactionTypes || []
+
+    console.log('form location:', location)
+    console.log('transaction details:', transaction, formData)
+    console.log('account details:', account, formData)
+
+
+    const handleInputChange = (e) => {
+        const { name, value } = e.target
+        setFormData((prevFormData) => ({
+            ...prevFormData,
+            [name]: value
+        }))
+    }
+
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+        if (isEditMode) {
+            console.log('formData:', formData)
+            console.log('edit mode')
+            try {
+
+                const response = await updateTransaction(transaction._id, formData, authToken)
```

Comment:
Again, we need error & loading state handling. You can apply your `useFetch` abstraction here!